platform/include/transport_interface.h

```diff
@@ -19,21 +19,147 @@
+ * @snippet this define_transportinterface
+ * <br>
+ *
+ * The following steps give guidance on implementing the transport interface:
+ *
+ * -# Implementing the @ref NetworkContext_t<br><br>
+ * @snippet this define_networkcontext
+ * <br>
+ * @ref NetworkContext_t is the incomplete type <b>struct NetworkContext</b>.
+ * The implemented struct NetworkContext should contain all of the information
+ * that is needed to receive and send data with the @ref TransportRecv_t
+ * and the @ref TransportSend_t implementations.<br>
+ * struct NetworkContext is typically implemented with the TCP socket context
+ * and a TLS context.<br><br>
+ * <b>Example code:</b>
+ * @code{c}
+ * struct NetworkContext
+ * {
+ *     struct MyTCPSocketContext tcpSocketContext;
+ *     struct MyTLSContext tlsContext;
+ * };
+ * @endcode
+ * <br>
+ * -# Implementing @ref TransportRecv_t<br><br>
+ * @snippet this define_transportrecv
+ * <br>
+ * This callback lets the application know that the library is ready to receive
+ * data over the network. Connection with the server of interest should have
+ * been made before invoking any library routines that receive data over the
+ * network.
+ * The @ref TransportRecv_t is typically implemented by calling directly the
+ * TLS layer function to receive data. If one is transferring over plaintext TCP,
+ * without TLS, then a call directly to the TCP layer function to receive data is
+ * made.
+ * @ref TransportRecv_t may be invoked multiple times by the library if less
+ * bytes than were requested to receive are returned.
+ * <br><br>
+ * <b>Example code:</b>
+ * @code{c}
+ * int32_t myNetworkRecvImplementation( const NetworkContext_t * pNetworkContext,
+ *                                      void * pBuffer,
+ *                                      size_t bytesToRecv )
+ * {
+ *     int32_t bytesReceived = 0;
+ *     bytesReceived = mySocketRecv( pNetworkContext->tcpSocketContext,
+ *                                   pBuffer,
+ *                                   bytesToRecv,
+ *                                   MY_SOCKET_TIMEOUT );
+ *     if( bytesReceived < 0 )
+ *     {
+ *         // Handle socket error.
+ *     }
+ *     // Handle other cases.
+ *
+ *     return bytesReceived;
+ * }
+ * @endcode
+ * <br>
+ * -# Implementing @ref TransportSend_t<br><br>
+ * @snippet this define_transportsend
+ * <br>
+ * This callback lets the application know that the library is ready to send
+ * data over the network. Connection with the server of interest should have
+ * been made before invoking any library routines that send data over the
+ * network.
+ * The @ref TransportSend_t is typically implemented by calling directly the
+ * TLS layer function to send data. If one is transferring over plaintext TCP,
+ * without TLS, then a call directly to the TCP layer function to send data is
+ * @ref TransportSend_t may be invoked multiple times by the library if less
+ * bytes than were requested to send are returned.
```

Comment:
I think a word is missing here. Also "less" -> "fewer"
```suggestion
 * without TLS, then a call is made directly to the TCP layer function to send data.
 * @ref TransportSend_t may be invoked multiple times by the library if fewer
 * bytes are returned than were requested to send.
```

Reply:
Thanks for catching. And I am ok with "fewer".